### ISSUE TITLE
False positive 'unsubscriptable-object' reported for `Enum.__members__` attribute

### ISSUE THREAD

**Issue by benburk** [PyCQA/pylint#4358]
### Steps to reproduce

```python
from enum import Enum
class Foo(Enum):
    BAR = 1
Foo.__members__['BAR']  # <-- error: Value 'Foo.__members__' is unsubscriptable
```

### Current behavior

Final line alerts "Value '`Foo.__members__`' is unsubscriptable"

### Expected behavior

No error

### pylint --version output
Result of `pylint --version` output:

```
pylint 2.7.4
astroid 2.5
Python 3.9.2 (default, Mar  3 2021, 11:58:52) 
[Clang 10.0.0 ]
```

**Comment by hippo91** [comment]
@benburk thanks for the report. It seems that @nelfin is working on a fix through PyCQA/astroid#941.
